src/Chapter3.hs

```diff
@@ -373,6 +381,27 @@ after the fight. The battle has the following possible outcomes:
    doesn't earn any money and keeps what they had before.
 
 -}
+data Unit = Unit
+    { health :: Int
+    , attack :: Int
+    , gold :: Int
+    }
+    deriving (Show)
+
+type Knight = Unit
+type Monster = Unit
+type CurrentUnit = Unit
+type AttackedUnit = Unit
```

Comment:
Yeah this one makes a lot of sense. I could've used this instead of defining a sum type for the type of entity or unit. It would've made things much easier for me to check an entity's type.